cassandra/cassandra_main.py

```diff
@@ -82,29 +82,30 @@ def main(argvals):
     for thread in component_threads:
         thread.join()
 
-    # Check to see if any of the components failed
-    fail = 0
+    # Check to see if any of the components failed, and that the RAB
+    # is still running
+    nfail = 0
     for component in component_list:
-        if component.status != 1:
+        if component.status != 1 or (type(component).__name__ == 'RAB' and component.status != 0):
```

Comment:
@calebbraun Instead of `type`, should we use `isinstance` instead?  That would future-proof us a bit, in case we ever decide to subclass the RAB for some reason.

Reply:
I was wondering this as well, but couldn't think of how to use `isinstance` without importing `rab.py` which leads to other troubles when we are not running mp.

Reply:
Turns out the logic here is wrong.  It fails the RAB because `component.status != 1`, and the other side of the `or` never even gets evaluated.  I'll fix it up and maybe try replacing `type` with `isinstance` along the way.

Reply:
Oh right, duh. Thanks!

Reply:
I decided the juice wasn't worth the squeeze on testing the component class (basically, we can't load any of the modules that import `mpi4py` in SP mode because we can't be sure users running in that mode have `mpi4py` installed, so the logic becomes sort of a pain.)  It was easiest just to take advantage of the RAB always being first in the list.

Reply:
(Looks like I independently rediscovered the same problem you were having.  It's almost like we could save some time if we, like, talked to each other or something.)